tools/javadoc/javadoc.bzl

```diff
@@ -46,6 +50,11 @@ def _javadoc_library(ctx):
         "-quiet",
     ]
 
+    # If not using `root_packages`, we'll roll up `srcs` using `args` so that
+    # we handle (a) tree artifacts and (b) very long argument lists.
+    args = ctx.actions.args()
+    args.use_param_file(param_file_arg = "@%s")
```

Comment:
I think you need `use_always = True` too, right?

Reply:
As I understand it, it depends. Without `use_always`, Bazel will expand short arg lists on the command-line and use the args file only for long arg lists. I don't know enough about Blaze/Bazel internals to have a strong opinion, but I have the impression that Google prefers to generate parameter files only when necessary.

It's a moot point w/ the latest revision, regardless. :)

Reply:
I see.